src/plone/restapi/serializer/__init__.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+from Acquisition import aq_inner
+from Acquisition import aq_parent
+from Products.CMFCore.interfaces import IContentish
+from Products.CMFCore.interfaces import IFolderish
+from Products.CMFPlone.interfaces import IPloneSiteRoot
+
+from plone.app.contenttypes.interfaces import ICollection
+from plone.restapi.interfaces import ISerializeToJson
+from plone.restapi.interfaces import IFieldSerializer
+from plone.restapi.serializer.converters import json_compatible
+from plone.dexterity.utils import iterSchemata
+
+from zope.site.hooks import getSite
+from zope.interface import implementer
+from zope.component import adapter
+from zope.component import getMultiAdapter
+from zope.component import queryMultiAdapter
+from zope.schema import getFields
+
+
+def lookup_field_serializer(context, field):
+    serializer = queryMultiAdapter((field, context, context.REQUEST),
+                                   IFieldSerializer,
+                                   name=field.interface.__identifier__)
```

Comment:
I've refactored the code and adde comments. I have also dropped the simple field name lookup, because I don't think it makes sense.
There is a standard adapter for a type of fields and a specific named adapter with the full dottedname. The full dottedname prevents naming conflicts of fields on multiple behaviors. I think that's enough of dependency injection entry points.